types.d.ts

```diff
@@ -1644,11 +1644,15 @@ declare class Compilation {
 		module: RuntimeModule,
 		chunkGraph?: ChunkGraph
 	): void;
+
+	/**
+	 * If `module` is passed, `loc` and `request` must also be passed.
+	 */
 	addChunkInGroup(
 		groupOptions: string | ChunkGroupOptions,
-		module: Module,
-		loc: DependencyLocation,
-		request: string
+		module?: Module,
+		loc?: SyntheticDependencyLocation | RealDependencyLocation,
```

Comment:
hmm, this seems to have been caused by the generation script. union normalization?

Reply:
yes, typescript isn't the best in keeping names. An optional argument adds `undefined` as possible value. So `DependencyLocation | undefined` is normalized to `SyntheticDependencyLocation | RealDependencyLocation | undefined` and for print `loc?` the `undefined` is omitted again => `SyntheticDependencyLocation | RealDependencyLocation`. But that doesn't matter.